flexget/plugins/sites/jackett.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+from past.utils import old_div
+from future.moves.urllib.parse import urlencode, quote_plus
+
+import logging
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.search import normalize_unicode
+import feedparser
+
+__author__ = 'davst'
+
+log = logging.getLogger('jackett')
+
+
+class Jackett(object):
+    """
+    Jackett search plugin
+    Based on the newznab plugin
+    Provide either url or website + apikey, and a category
+
+    Config examples::
+
+        Jackett:
+          website: https://website
+          apikey: xxxxxxxxxxxxxxxxxxxxxxxxxx
+          category: movie
+          
+        -- or --
+        
+        Jackett:
+          url: "http://website/api?apikey=xxxxxxxxxxxxxxxxxxxxxxxxxx&t=movie&extended=1"
+          category: movie
+
+    Category is any of: movie, tv, search (default)
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'category': {'type': 'string', 'enum': ['movie',  'tv', 'search']},
+            'url': {'type': 'string', 'format': 'url'},
+            'website': {'type': 'string', 'format': 'url'},
+            'apikey': {'type': 'string'}
+        },
+        'oneOf': [
+            {'required': ['url']},
+            {'required': ['website', 'apikey']}
+        ],
+        'additionalProperties': False
+    }
+
+    def build_config(self, config):
+        log.debug(type(config))
+
+        if 'url' not in config:
+            if 'apikey' in config and 'website' in config:
+                params = {
+                    't': config['category'],
```

Comment:
This will crash if not category has been chosen since you didn't define a default for `category`